projects/plugins/jetpack/sal/class.json-api-site-base.php

```diff
@@ -766,7 +766,7 @@ private function user_can_view_post( $post ) {
 			$post_status_obj = get_post_status_object( $post->post_status );
 		}
 
-		$authorized = (
+		$authorized = null !== $post_status_obj && (
```

Comment:
While this fixes the direct issue of the warning, it masks a logic error in the code above.

Attachments have a post status of `inherit`. It's possible for an attachment to not have a parent (an unattached attachment). So, the logic with the post status `inherit` check above needs to be changed to ensure that we don't check the parent if there isn't a parent. Perhaps logic such as the following:

```
if ( 'inherit' === $post->post_status && 0 !== (int)$post->post_parent ) {
```

Could you update the logic to ensure that we are checking the post status of the current post and not the parent in the event that a parent doesn't exist?

Reply:
Thanks for the review @chrisbliss18. This definitely makes sense. I've implemented it in [2bba21e](https://github.com/Automattic/jetpack/pull/39727/commits/2bba21efa85d237d92ffabb6ab8bb98510431cc8).
Let me know what you think